Dockerfile

```
@@ -12,14 +12,11 @@ RUN apt-get update && apt-get install -y \
     && rm -rf /var/lib/apt/lists/*
 
-# Set working directory
-WORKDIR /app
-
 # Copy dependency list
 COPY requirements.txt .
 
-# Install Python packages
-RUN pip install --upgrade pip && pip install -r requirements.txt
+# Install Python packages with no cache
+RUN pip install --no-cache-dir --upgrade pip && pip install --no-cache-dir -r requirements.txt
```

Hadolint (2.12.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)